filament/backend/src/metal/MetalHandles.mm

```diff
@@ -267,42 +267,18 @@ BufferDescriptor b(nullptr, 0u, [](void* buffer, size_t size, void* user) {
 }
 
 MetalBufferObject::MetalBufferObject(MetalContext& context, uint32_t byteCount)
-        : byteCount(byteCount), buffer(std::make_shared<MetalBuffer>(context, byteCount)) {}
+        : HwBufferObject(byteCount), buffer(std::make_unique<MetalBuffer>(context, byteCount)) {}
 
 void MetalBufferObject::updateBuffer(void* data, size_t size, uint32_t byteOffset) {
     assert_invariant(byteOffset + size <= byteCount);
     buffer->copyIntoBuffer(data, size);
 }
 
 MetalVertexBuffer::MetalVertexBuffer(MetalContext& context, uint8_t bufferCount,
-            uint8_t attributeCount, uint32_t vertexCount, AttributeArray const& attributes,
-            bool bufferObjectsEnabled)
-    : HwVertexBuffer(bufferCount, attributeCount, vertexCount, attributes, bufferObjectsEnabled) {
+            uint8_t attributeCount, uint32_t vertexCount, AttributeArray const& attributes)
+    : HwVertexBuffer(bufferCount, attributeCount, vertexCount, attributes) {
     buffers.reserve(bufferCount);
-
-    if (bufferObjectsEnabled) {
-        // If BufferObjects are used, we don't need to make any allocations. Simply reserve enough
-        // "slots" for future calls to setVertexBufferObject.
-        buffers.resize(bufferCount);
-        return;
-    }
-
-    for (uint8_t bufferIndex = 0; bufferIndex < bufferCount; ++bufferIndex) {
-        // Calculate buffer size.
-        uint32_t size = 0;
-        for (auto const& item : attributes) {
-            if (item.buffer == bufferIndex) {
-                uint32_t end = item.offset + vertexCount * item.stride;
-                size = std::max(size, end);
-            }
-        }
-
-        std::shared_ptr<MetalBuffer> buffer = nullptr;
-        if (size > 0) {
-            buffer = std::make_shared<MetalBuffer>(context, size);
-        }
-        buffers.push_back(buffer);
-    }
+    buffers.resize(bufferCount);
```

Comment:
I think we can remove the call to `buffers.reserve` here.

Reply:
Good catch, removed.